languages/python/reference/exercise-concepts/robot-simulator.md

```diff
@@ -0,0 +1,62 @@
+# Concepts for robot-simulator
+
+[Example Implementation](https://github.com/exercism/python/blob/master/exercises/robot-simulator/example.py)
+
+
+## General
+
+- contstants: Are not enforced by the runtime, but are used via the convention of `<ALL_CAPS_NAMING>` to signal that these values are expected to remaine unchanged.  Perferably, constants are defined at a module level. The example solution uses the ALL_CAPS convention to define NORTH, SOUTH, EAST, and WEST constants. 
+
+- multiple assignment:  Python allows multiple assignment, assiging the items on the left of `=` _in order_ to the values on the right of `=`.  Equivelent to forming tuples of each side and then "unpacking" them.  This exercise solution uses multiple assignment for the contant values NORTH, EAST, SOUTH, WEST.
+
+- `range()` built-in type:  Represents an immutable sequence of numbers (or any object that implements the `__index__` magic method).  Used in the example to represnet the values from zero to 3 as assigned to NORTH, EAST, SOUTH, WEST. 
+
+- classes: The exercise objective is to define a `robot` type.  Tested methods are linked to a `robot` class.
+
+- objects: Creating different instances of the `robot` class with different data representing different starting positions and bearing are tested.
+
+- constructor: Customizing object initalization with actions and persisting data. The example uses a constructor (via `__init__` + `__new__`) to persist a `compass` object and x, y coordinates assigned to instance properties.
+
+- retun values:  Knowing that functions need not have explicit return statements or values but will return `none` if `return` is not specified.  Except for the two `@property` functions, all of the functions in the example omit an explicit `return` statments, and all return `none`.
```

Comment:
Spelling: "retun" -> "return", "statments" -> "statements". Also both occurrences of "none" should really be "None". Also it may be helpful to say "`@property`-decorated functions` for clarity. I find it helps to visualize an implicit `return None` at the bottom of every function I start to write.